packages/core/src/queries/connector.ts

```diff
@@ -0,0 +1,19 @@
+import { Connector, ConnectorDBEntry, Connectors, ConnectorType } from '@logto/schemas';
+import { sql } from 'slonik';
+
+import { buildInsertInto } from '@/database/insert-into';
+import pool from '@/database/pool';
+import { convertToIdentifiers } from '@/database/utils';
+
+const { table, fields } = convertToIdentifiers(Connectors);
+
+export const findConnectorByIdAndType = async (id: string, type: ConnectorType) =>
```

Comment:
need to align the usage: in `packages/core/src/queries/application.ts:18`, `findApplicationById` simply uses `.one()` as it assume the caller will catch "not found" error. we'd better to align one usage for `findXyz()`:

1. use `.one()` for all `findXyz()`, and wrap it with a function like `optional(findXyz())` to return optional value
2. use another name for optional functions
3. use `.maybeOne()` for all `findXyz()`

i'm ok with every approach. any ideas? cc @simeng-li @IceHe @darcyYe

Reply:
ideally, zero or one row of results will be found which should be ensured by DB schema design. thus option 3 is preferred on my side.

Reply:
I prefer option 1.

If so, we don't have to catch "not found" errors all the time and we can also throw an error as needed.

When `findMany` cannot find anything return empty list rather than `optional`.

Reply:
I do also have a maybeOne query used [here](https://github.com/logto-io/logto/blob/dd7a38536377aecba03b1bffa470f94abf503090/packages/core/src/queries/resources.ts#L33). 
In my case, I need to handle the error explicitly from an upper scope.   If no resources found, throw an oidc invalid target error. 
If go with option 1, we need to capture the error manually and throw a customized error.  
So it all determines on how we gonna handle not found errors.

Reply:
if it is a common and expected case for data not found, I think use maybeOne is fine?

Reply:
I prefer option 3, is the controller's job to handle the existance problem, it should explicity check this, in most cases, throw 404 exception, 422 in some cases, and other unexpected cases, 500 (maybe with an error alert) is better than 404. Treat all DB's not found as a http-not-found exception could cause potencial problems.

This is beyond this PR, so I created LOG-1300 to track, and can I ignore this probem in this PR to unblock development?

Reply:
Just curious, what problem will cause exactly? We can discuss it later